devicetypes/stelpro/stelpro-ki-thermostat.src/stelpro-ki-thermostat.groovy

```diff
@@ -291,22 +296,27 @@ def zwaveEvent(sensormultilevelv3.SensorMultilevelReport cmd) {
 
 		temp = map.value
 		// The specific values checked below represent ambient temperature alarm indicators
-		if (temp == "32765") {			// 0x7FFD
+		if (temp == 0x7ffd) {
 			map.name = "temperatureAlarm"
 			map.value = "freeze"
-			map.unit = ""
-		} else if (temp == "32767") {		// 0x7FFF
+			map.unit = null
```

Comment:
Does setting `unit` to `null` work? Another option might be to do `map.remove('unit')` if the goal is to get rid of the unit. I'm  not sure if one way is preferable over another if they both work.

Reply:
Actually I think it'd probably be more clear to just fully initialize the event for each specific case rather than setting value,  unit and name up front and overriding certain values. It might be a little extra code but it won't be much and I think it will be easier to understand.

Reply:
Setting unit to null does work!

So far, since the logic was getting big, I moved the code out of parse() and put it in handleTemperature(). I am actually going to change the way parse() builds events because there will be a few times I want to pass more than one thing back from parse().